### ISSUE TITLE
Location of beacin is not of beacon but current location

### ISSUE THREAD

**Issue by jcurl** [seemoo-lab/AirGuard#4]
I'm using Android AirGuard 1.0.4 installed by F-Droid.

Thank you for an interesting app, already more beacons than I expected.

When I choose a beacon, it shows a map, but instead of showing the location at the time the beacon was identified, it shows my current location.

**Comment by Sn0wfreezeDev** [comment]
Thank you for the report. We will look into this

**Comment by Trup3s** [comment]
@jcurl Do you have the location in the app settings enabled? Does this behavior happen for every device you select?

**Comment by jcurl** [comment]
yes, the behaviour is the same for all devices. Thanks for taking the time to respond.

Steps to reproduce:
1. From the dashboard, choose Devices found
2. Select a device (e.g. something from a day where it's known I'm in a different position currently)
3. The device MAC shows up on the top, a map is shown and the map position is my current position, not the position at the time the beacon was identified. I would have expected the map to be centered on the beacon, not my position.

See screenshot;
![Screenshot_20210921-171648_AirGuard](https://user-images.githubusercontent.com/1563519/134198727-f4139ef9-fd73-47ec-9706-6b7d59f2e8fa.png)

After playing around a little longer, there are some beacons when I request do actually center on the likely location of the beacon (or specifically, my location at the time the beacon was identified).

See screenshot:
![Screenshot_20210921-171920_AirGuard](https://user-images.githubusercontent.com/1563519/134199438-98a55b2c-fe45-4f46-922b-5620e1c8df32.png)

It appears to occur more often when the number of beacons is more than 1 (but not always, there's one case I see 2 beacons and it's correctly shown).

How do I interpret the case when the map doesn't show a beacon?

**Comment by Trup3s** [comment]
> How do I interpret the case when the map doesn't show a beacon?  

In your case the most likely reason is that your phone didn't get an accurate enough location (>100m) and saved the beacon without any location data. This can happen relatively easily when you're in a building like your screenshot suggest.  

We should probably add a warning that some beacons have been captured without any location data to prevent such issues.  

Can you confirm that this issue persists when you're outside?

**Comment by jcurl** [comment]
You may be right after I realized that some beacons had positions, while others don't. 

It is hard to verify, as I myself do not have any devices I know of, to test for tracking.

**Comment by Trup3s** [comment]
No problem! But there are surprisingly many iPhones or other FindMy devices that can be found during a short walk through the city ;)
